File: microservicio-clinica-nest/src/modules/clinical-records/dto/update-clinical-record-in.dto.ts
import { IsString, IsOptional } from 'class-validator';
import { ApiPropertyOptional } from '@nestjs/swagger';


export class UpdateClinicalRecordInDto  {

 @ApiPropertyOptional({ 
    example: 'III',
    description: 'Nueva etapa del cáncer (ej: IIA, III, IV)',
  })
  @IsString({ message: 'La etapa debe ser texto' })
  @IsOptional()
  stage?: string;

  @ApiPropertyOptional({ 
    example: 'Quimioterapia de segunda línea + inmunoterapia',
    description: 'Nuevo protocolo de tratamiento',
  })
  @IsString({ message: 'El protocolo de tratamiento debe ser texto' })
  @IsOptional()
  treatmentProtocol?: string;


}

File: autenticación/src/main/java/com/creators/autenticacion/models/dto/clinicalRecord/UpdateClinicalRecordInDto.java
package com.creators.autenticacion.models.dto.clinicalRecord;

import com.fasterxml.jackson.annotation.JsonInclude;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
@Schema(description = "DTO para actualizar la evolución de una historia clínica (stage y/o tratamiento)")

public class UpdateClinicalRecordInDto {

    @Schema(description = "Nueva etapa del cáncer (ej: IIA, III, IV)", example = "III")
    private String stage;

    @Schema(description = "Nuevo protocolo de tratamiento", example = "Quimioterapia de segunda línea + inmunoterapia")
    private String treatmentProtocol;

}
